internal/collector/event.go

```diff
@@ -95,12 +94,25 @@ func (collector *EventCollector) increaseEventsTotal(event *v1.Event, nbNew floa
 	}).Add(nbNew)
 }
 
-func beforeLastEvent(t time.Time, ev *v1.Event) bool {
-	if ev.Series != nil && !ev.Series.LastObservedTime.IsZero() {
-		return t.Before(ev.Series.LastObservedTime.Time)
+func beforeLatestEvent(t time.Time, ev *v1.Event) bool {
+	eventTimes := []time.Time{
+		ev.FirstTimestamp.Time,
```

Comment:
what if event is nil?

Reply:
It should never be nil as the Events are provided by the informer event handler.